src/Build/Evaluation/Expander.cs

```diff
@@ -1517,6 +1513,10 @@ private static object LookupProperty(IPropertyProvider<T> properties, string pro
                 else
                 {
                     propertyValue = property.EvaluatedValueEscaped;
+                    if (!string.IsNullOrEmpty(property.EvaluatedValueEscaped) && property.IsEnvironmentProperty)
```

Comment:
we should find some way to not repeat `property.EvaluatedValueEscaped` three times in this else block. If propertyValue is an object, let's perhaps do something like this:

```
if (propertyValue is { Length: > 0 } value && property.IsEnvironmentProperty) ...
```